server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/ElasticSearchMailboxModule.java

```diff
@@ -100,6 +104,16 @@ private ElasticSearchSearcher createMailboxElasticSearchSearcher(Client client,
             MailboxElasticSearchConstants.MESSAGE_TYPE);
     }
 
+    @Provides
+    @Singleton
+    private ElasticSearchHealthCheck createElasticSearchHealthCheck(Client client, ElasticSearchConfiguration configuration) {
+        Set<IndexName> indices = new HashSet<>();
+        indices.add(configuration.getIndexMailboxName());
```

Comment:
We can inject this set in a dynamic way using the guice multi-binder utility. Thus each module becomes responsible of declaring which indexes it uses and avoid a central point of knowledge.

In this class configure method:


```
Multibinder.newSetBinder(binder(), IndexName.class)
    .addBinding().to(configuration.getIndexMailboxName());
```

Idem we can bind the `configuration.getIndexQuotaRationName` in `ElasticSearchQuotaSearcherModule`